unstructured/partition/email.py

```diff
@@ -1,510 +1,429 @@
-                    name=addr.display_name or addr.username,  # type: ignore
-                    text=addr.addr_spec,  # type: ignore
-                )
-            )
+class _AttachmentPartitioner:
+    """Partitions an attachment to a MSG file."""
 
-    elements: list[Element] = []
-
-    for msg_field, msg_value in msg.items():
-        if msg_field in {"To", "Bcc", "Cc"}:
-            append_address_header_elements(msg_value, Recipient)
-        elif msg_field == "From":
-            append_address_header_elements(msg_value, Sender)
-        elif msg_field == "Subject":
-            elements.append(Subject(text=msg_value))
-        elif msg_field == "Received":
-            elements += _parse_received_data(msg_value)
-        elif msg_field == "Message-ID":
-            elements.append(MetaData(name=msg_field, text=_strip_angle_brackets(str(msg_value))))
-        else:
-            elements.append(MetaData(name=msg_field, text=msg_value))
+    def __init__(self, attachment: EmailMessage, ctx: EmailPartitioningContext):
+        self._attachment = attachment
+        self._ctx = ctx
 
-    return elements
+    @classmethod
+    def iter_elements(
+        cls, attachment: EmailMessage, ctx: EmailPartitioningContext
+    ) -> Iterator[Element]:
+        """Partition an attachment MIME-part from a MIME email message (.eml file)."""
+        return cls(attachment, ctx)._iter_elements()
 
+    def _iter_elements(self) -> Iterator[Element]:
+        """Partition the byte-stream in the attachment MIME-part into elements.
 
-def _strip_angle_brackets(data: str) -> str:
-    """Remove angle brackets from the beginning and end of the string if they exist.
+        Generates zero elements if the attachment is not partitionable.
+        """
+        # -- `auto.partition()` imports this module, so we need to defer the import to here to
+        # -- avoid a circular import.
+        from unstructured.partition.auto import partition
 
-    Returns:
-    str: The string with surrounding angle brackets removed.
+        file = io.BytesIO(self._file_bytes)
 
-    Example:
-    >>> _strip_angle_brackets("<example>")
-    'example'
-    >>> _strip_angle_brackets("<another>test>")
-    'another>test'
-    >>> _strip_angle_brackets("<<edge>>")
-    '<edge>'
-    """
-    return re.sub(r"^<|>$", "", data)
+        # -- partition the attachment --
+        try:
+            elements = partition(
+                file=file,
+                metadata_filename=self._attachment_file_name,
+                metadata_last_modified=self._ctx.metadata_last_modified,
+                **self._ctx.partitioning_kwargs,
+            )
+        except UnsupportedFileFormatError:
+            # -- indicates `auto.partition()` has no partitioner for this file-format;
+            # -- silently skip the attachment
+            return
+
+        for e in elements:
+            e.metadata.attached_to_filename = self._attached_to_filename
+            yield e
```

Comment:
Just noticing that you define `e.metadata.attached_to_filename` directly here instead of using the `.update()` method. Doesn't need to change, but any reason why?

Reply:
`.update()` requires a whole `ElementMetadata()` object to be passed, like "update yourself from this other `ElementMetadata` instance". That can be handy when you're updating multiple fields or collecting the fields to update from multiple places, but isn't necessary here.